src/demo/models/pond.js

```diff
@@ -33,23 +35,48 @@ const predictAllFish = state => {
 };
 
 const arrangeFish = fishes => {
+  let fishPositions = formatArrangement();
+
   fishes.forEach(fish => {
-    // Pick a random side of the bot UI.
-    const side = randomInt(0, 1);
-
-    // Generate a location for the fish on that side.
-    const xBounds = [
-      {minX: 0, maxX: 360 - constants.fishCanvasWidth / 2},
-      {minX: 510, maxX: constants.canvasWidth - constants.fishCanvasWidth}
-    ];
-    const x = randomInt(xBounds[side].minX, xBounds[side].maxX);
-
-    // Don't put fish at the very bottom of the pond because of UI.
-    const bottomAreaHeight = 80;
-    const y = randomInt(
-      0,
-      constants.canvasHeight - constants.fishCanvasHeight / 2 - bottomAreaHeight
-    );
+    const pos = fishPositions.shift();
+    const x = pos[0] * 140 - 50;
+    const y = pos[1] * 150;
+
     fish.setXY({x, y});
   });
 };
+
+// Describes the 20 possible fish positions on the screen, where the value describes
+// that position's priority. 0 will be filled first, then 1, etc.
+const arrangement = [
+  [2, 1, 0, 0, 0, 1, 2],
+  [2, 1, 0, 0, 0, 1, 2],
+  [2, 1, 0, null, 0, 1, 2]
+];
+
+// Reformats the arrangement constant into a 1-dimensional array of x-y coordinates,
+// ordered in priority order (e.g., the spots to fill first appear first in the array).
+const formatArrangement = () => {
+  let intermediateArr = [];
+  arrangement.forEach((row, rowIdx) => {
+    row.forEach((col, colIdx) => {
+      if (typeof col !== 'number') {
+        return;
+      }
+
+      if (!intermediateArr[col]) {
+        intermediateArr[col] = [];
+      }
+
+      intermediateArr[col].push([colIdx, rowIdx]);
+    });
+  });
+
+  // Flatten nested intermediateArr into a 1-dimensional array.
```

Comment:
Can you elaborate here what the 1-dimensional array represents (e.g. the coordinates in the order they should be filled)?